### PR TITLE
Fix README links to docs

### PR DESCRIPTION
**Pull request by gsheni**
### Pull Request Description
- Fix the development link in the README to the correct link

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,7 +81,7 @@ customer_id
 
 [5 rows x 69 columns]
 ```
-We now have a feature vector for each customer that can be used for machine learning. See the [documentation on Deep Feature Synthesis](https://docs.featuretools.com/automated_feature_engineering/afe.html) for more examples. 
+We now have a feature vector for each customer that can be used for machine learning. See the [documentation on Deep Feature Synthesis](https://docs.featuretools.com/automated_feature_engineering/afe.html) for more examples.
```
Comment (rwedge) [inline README.md:84]: Can you fix this link as well?

File: docs/source/changelog.rst [hunk 2]
```diff
@@ -15,6 +15,7 @@ Changelog
         * Add source code links to API reference (:pr:`862`)
         * Update links for testing Dask/Spark integrations (:pr:`867`)
         * Update release documentation for featuretools (:pr:`868`)
+        * Fix README link to development (:pr:`872`)
```
Comment (rwedge) [inline docs/source/changelog.rst:18]: Can you start a new "Future Release" section above this one?

File: README.md (hunk 1 above)
Comment (gsheni) [reply README.md]: Fixed

File: docs/source/changelog.rst (hunk 2 above)
Comment (gsheni) [reply docs/source/changelog.rst]: Added

**Pull request merged**